example-scene/src/turrets.ts

```diff
@@ -0,0 +1,138 @@
+import { Entity, engine, Transform, MeshRenderer, Raycast, RaycastQueryType, Schemas, RaycastResult, MeshCollider, ColliderLayer } from "@dcl/sdk/ecs"
+import { Vector3, Quaternion } from "@dcl/sdk/math"
+import { registerRaycastHit } from "./raycastHits"
+
+const TurretSegment = engine.defineComponent("TurretArticulatedSegment", {
+  index: Schemas.Float,
+  current: Schemas.Float
+})
+
+export function addTurretSegment(parent: Entity, index: number) {
+  const segment = engine.addEntity()
+  TurretSegment.create(segment, { index })
+  Transform.create(segment, { parent, position: { x: 0, y: 0, z: 1 } })
+  const cube = engine.addEntity()
+  Transform.create(cube, { parent: segment, scale: { x: 0.5 / index, y: 0.5 / index, z: 1 }, position: { x: 0, y: 0, z: 0.5 } })
+  MeshRenderer.setBox(cube)
+  return segment
+}
+
+export function initRaycastTurrets(parent: Entity) {
+  const floor = engine.addEntity()
+  Transform.create(floor, {
+    parent,
+    position: { x: 8, y: 0, z: 8 },
+    scale: { x: 16, y: 0.01, z: 16 },
+  })
+  MeshRenderer.setBox(floor)
+  const collisionMask = ColliderLayer.CL_PHYSICS
+  MeshCollider.setBox(floor, collisionMask)
+
+
+  {
+    const parentEntity = engine.addEntity()
+    const parentTransform = Transform.create(parentEntity, { parent })
+
+    parentTransform.position.x = 12
+    parentTransform.position.y = 5
+    parentTransform.position.z = 12
+
+    const segment1 = addTurretSegment(parentEntity, 1)
+    const segment2 = addTurretSegment(segment1, 2)
+    const segment3 = addTurretSegment(segment2, 3)
+    const segment4 = addTurretSegment(segment3, 4)
+    const segment5 = addTurretSegment(segment4, 6)
+
+    Raycast.create(segment5, {
+      originOffset: Vector3.scale(Vector3.Forward(), 1.1),
+      direction: {
+        $case: 'localDirection',
+        localDirection: Vector3.Forward()
+      },
+      continuous: true,
+      maxDistance: 999,
+      queryType: RaycastQueryType.RQT_HIT_FIRST
+    })
+  }
+
+  {
+    const parentEntity = engine.addEntity()
+    const parentTransform = Transform.create(parentEntity, { parent })
+
+    parentTransform.position.x = 8
+    parentTransform.position.y = 8
+    parentTransform.position.z = 8
+
+    const segment1 = addTurretSegment(parentEntity, 1)
+    const segment2 = addTurretSegment(segment1, 9)
+    const segment3 = addTurretSegment(segment2, 3)
+    const segment4 = addTurretSegment(segment3, 4)
+    const segment5 = addTurretSegment(segment4, 4)
+
+    Raycast.create(segment5, {
+      originOffset: Vector3.scale(Vector3.Forward(), 1.1),
+      direction: {
+        $case: 'globalTarget',
+        globalTarget: Vector3.Zero()
+      },
+      continuous: true,
+      maxDistance: 999,
+      queryType: RaycastQueryType.RQT_HIT_FIRST
+    })
```

Comment:
_Style suggestion_
I like this. The `Raycast` definitions are much more readable than the `Transform` ones, this is what I had in mind.

How about reordering the parameters a bit, like so...

```
Raycast.create(segment5, {
  queryType: RaycastQueryType.RQT_HIT_FIRST,
  continuous: true,
  maxDistance: 999,
  originOffset: Vector3.scale(Vector3.Forward(), 1.1),
  direction: {
    $case: 'globalTarget',
    globalTarget: Vector3.Zero()
  },
})
```

... this is closer to the mental process in my head when I go through the object. Also, the first 3 parameters are trivial to understand, while the last 2 require playing with vectors in my head.